src/lwc/rd2EntryFormDonorSection/__tests__/rd2EntryFormDonorSection.test.js

```diff
@@ -0,0 +1,251 @@
+    let element;
+
+    beforeEach(async () => {
+        getRecurringData.mockResolvedValue(recurringDataContactResponse);
+        element = createDonorSection({ recordId: FAKE_RD_ID });
+        getObjectInfo.emit(recurringDonationObjectInfo, ({objectApiName}) => {
+            return objectApiName === RECURRING_DONATION_OBJECT.objectApiName
+        });
+        await flushPromises();
+    });
+
+    afterEach(() => {
+        clearDOM();
+        jest.clearAllMocks();
+    })
+
+    it('loads with donor type set to contact', () => {
+        assertDonorTypeFieldValue(element, 'Contact');
+    });
+
+    it('notifies parent component of donor type change', () => {
+        expect(mockHandleDonorTypeChange).toHaveBeenCalled();
+        const lastEventDetail = getLastEventDetail(mockHandleDonorTypeChange);
+        expect(lastEventDetail).toBe('Contact');
+    });
+});
+
+
+describe('existing recurring account donation', () => {
+    let element;
+
+    beforeEach(async () => {
+        getRecurringData.mockResolvedValue(recurringDataAccountResponse);
+
+        element = createDonorSection({ recordId: FAKE_RD_ID });
+
+        getObjectInfo.emit(recurringDonationObjectInfo, ({objectApiName}) => {
+            return objectApiName === RECURRING_DONATION_OBJECT.objectApiName
+        });
+
+        await flushPromises();
+    });
+
+    afterEach(() => {
+        clearDOM();
+        jest.clearAllMocks();
+    });
+
+    it('loads with donor type set to Account', () => {
+        expect(getRecurringData).toHaveBeenCalled();
+        assertDonorTypeFieldValue(element, 'Account');
+    });
+
+    it('clears values in Account lookup when donor type changed to Contact', async () => {
+        changeDonorType(element, 'Contact');
+        assertDonorTypeFieldValue(element, 'Contact');
+
+        await flushPromises();
+
+        expect(getAccountLookup(element)).toBeNull();
+    });
+});
+
+describe('new donation from account', () => {
+    let element;
+
+    beforeEach(async () => {
+        getRecurringData.mockResolvedValue(recurringDataAccountResponse);
+
+        element = createDonorSection({ parentId: FAKE_ACCOUNT_ID, parentSObjectType: 'Account' });
+
+        getObjectInfo.emit(recurringDonationObjectInfo, ({objectApiName}) => {
+            return objectApiName === RECURRING_DONATION_OBJECT.objectApiName
+        });
+
+        await flushPromises();
+    });
+
+    afterEach(() => {
+        clearDOM();
+        jest.clearAllMocks();
+    });
+
+    it('alerts parent component of accountId', () => {
+        const lastEventDetail = getLastEventDetail(mockHandleAccountChange);
+        expect(lastEventDetail).toBe(FAKE_ACCOUNT_ID);
+    });
+
+    it('populates account lookup', () => {
+        const accountLookup = getAccountLookup(element);
+        expect(accountLookup).toBeTruthy();
+        expect(accountLookup.value).toBe(FAKE_ACCOUNT_ID);
+    });
+
+    it('clears account lookup on donor type change', async () => {
```

Comment:
I *think* this is testing the following AC, but I want to confirm: "When a user selects Account for Donor Type And they enter an Account And they switch back to Contact for Donor Type Then the Account value is cleared"

Reply:
I don't think I have that particular case here, stopped one donor type change short of it! Will add.

Reply:
Added!